tests/chainer_tests/functions_tests/loss_tests/test_crf1d.py

```diff
@@ -1,90 +1,119 @@
+import itertools
 import unittest
 
 import numpy
 
 import chainer
+from chainer import cuda
 from chainer import functions
+from chainer import gradient_check
 from chainer import testing
+from chainer.testing import attr
 
 
+@testing.parameterize(
+    {'lengths': [3, 3], 'batches': [2, 2, 2]},
+    {'lengths': [3, 2], 'batches': [2, 2, 1]},
```

Comment:
`lengths` is transpose of `batches`, and we can automatically calculate one from other. However, I wrote both of them by hand in order to avoid a bug in the test code.